### ISSUE TITLE
Huggingface model produces different embeddings

### ISSUE THREAD

**Issue by malteos** [allenai/specter#24]
Hi @armancohan 

the available SciDocs embeddings and the embeddings returned from the API do not match the embeddings derived from the Huggingface model. Are the model weights the same? Or any idea why this is the case? 

I'm using the following code:
```python
tokenizer = AutoTokenizer.from_pretrained('allenai/specter')
model = AutoModel.from_pretrained('allenai/specter')

papers = [{'paper_id': 'A', 'title': 'BERT', 'abstract': 'We introduce a new language representation model called BERT'}]

# concatenate title and abstract
title_abs = [d['title'] + tokenizer.sep_token + (d.get('abstract') or '') for d in papers]
# preprocess the input
inputs = tokenizer(title_abs, padding=True, truncation=True, return_tensors="pt", max_length=512)
model.eval()
with torch.no_grad():
    result = model(**inputs)
# take the first token in the batch as the embedding
embeddings = result.last_hidden_state[:, 0, :]

pred_embed = embeddings.numpy()
# => array([[-8.39614451e-01,  1.14658070e+00, -5.10430574e-01, ...

# Embeddings via API
response = requests.post("https://model-apis.semanticscholar.org/specter/v1/invoke", json=papers)

true_embed = np.array([response.json()['preds'][0]['embedding']])
# =>  array([[-3.30647826e-02, -4.52146339e+00,  4.61816311e-01, ...

(true_embed==pred_embed).all()
# => False
```

**Comment by armancohan** [comment]
This is expected as the Huggingface model was re-trained from scratch. However, the results are very similar.
If curious about the numbers, please see the [README](https://docs.google.com/document/d/1ueSNBo_Zvs3dCSsm6AepZ7YRcAXdipjBVp1B8868gu4/edit#heading=h.ugutkur834) for the expected huggingface results and compare them with original results reported on our paper.

**Comment by malteos** [comment]
Thanks for the clarification. I missed that part in the README. 

With "re-trained from scratch" you refer to training from the SciBERT checkpoint, right? And did you use the same paper triplets for training?

**Comment by armancohan** [comment]
Sure. 
Yes, to clarify, by "from scratch" I mean starting from scibert checkpoint and continue training using the triplet loss objective.
